### ISSUE TITLE
Parallelize R scripts

### ISSUE THREAD

**Issue by Bubblbu** [OpenKnowledgeMaps/Headstart#32]
**PLOS queries**

Is searchplos.r already parallelized? If not, would it improve the performance if we would use parallel `searchplos(start=i, limit=10)` calls with `i = seq(0, len_papers,  @10)`?

**Non-metric multidimensional scaling**

Instead of using `nmds {ecodist}` ([doc](http://www.inside-r.org/packages/cran/ecodist/docs/nmds)), we could use `par.nmds {parfossil}`([doc](http://www.inside-r.org/packages/cran/parfossil/docs/par.nmds)) which is the implementation of the _ecodist_ function with multi-core foreach-loops instead of the standard for-loops.

At least on my machine this reduced the runtime for nmds from ~15s to ~5s

**Comment by Bubblbu** [comment]
@sckott

**Comment by sckott** [comment]
So at least a while back PLOS was rate limiting, so the `searchplos()` function has rate limiting, see https://github.com/ropensci/rplos/blob/master/man-roxygen/plos.r#L21-L24 for details. 

It may have changed, I don't know. It's worth trying to parallelize and see what happens. 

PLOS search is a Solr service, so it's very flexible if you want to modify how you search and what's returned.

**Comment by pkraker** [comment]
I have now implemented par.nmds. Take a look at the runtimes on Open Knowledge Maps, should be much faster already. I think, we could use the R package for parallelization (doParallel), which I had to load in order for par.nmds to work, to also make parallel calls to searchplos: https://cran.r-project.org/web/packages/doParallel/vignettes/gettingstartedParallel.pdf As @sckott mentioned, Solr is pretty flexible, and also very efficient, so we'd have to check whether this acutally speeds up our request. Also, as @sckott said, we might run into problems with rate limiting by PLOS.
